src/compiler/checker.ts

```diff
@@ -44826,7 +44826,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             let meaning = SymbolFlags.None;

```

Comment:
I can't put a comment on it, but the line above has an `isHeritageClauseElementIdentifier`; I feel like this function needs to be renamed to something that indicates that the name might _not_ be in an heritage clause.

That, or we fix the helper to check `isPartOfTypeNode`, then create a whole new block for this non-type-space thing below?

Reply:
Oh, `isHeritageClauseElementIdentifier` is literally only used in this one place. I vote for renaming it, maybe `isNameOfExpressionWithTypeArguments` or something?

Reply:
I think `isNameOfExpressionWithTypeArguments` is a good name because the function checks if the name is an `ExpressionWithTypeArguments`, so I support that :)

Reply:
I think `isInNameOfExpressionWithTypeArguments` might be more consistent with other helpers (note the extra "in"), since it also applies to dotted names like the left side of `Foo.Bar<Baz>`.